### ISSUE TITLE
Bitbucket SSH Keys in RSA format deprecated

### ISSUE THREAD

**Issue by mfabricanti** [argoproj/argo-cd#13749]
Hello there,

Bitbucket is warning us that some SSH Keys in RSA format will be deprecated soon (Jun 20).
Details here: [ssh-host-key-changes](https://bitbucket.org/blog/ssh-host-key-changes)

Should we worry since we have only one SSH RSA Key in CM argocd-ssh-known-hosts-cm for bitbucket.org?

Thanks in advance

**Comment by afreemanio** [comment]
Can confirm I ran into this issue earlier today. Following first time setup instructions from bitbucket and the docs, when I entered a newly created SSH key in the format of `ed25519 -b 4096` as they recommend, using argocd 2.3, I got the following error: ```error testing repository connectivity: ssh: handshake failed: knownhosts: key mismatch"```

Fixed it by following the instructions on their [blogpost ](https://bitbucket.org/blog/ssh-host-key-changes) for creating new known_hosts (`ssh-keygen -R bitbucket.org && curl https://bitbucket.org/site/ssh`), and copied the output into the argocd certficiates ("Add ssh known hosts") to fix the issue. 

Obviously my problem was not with RSA keys specifically, but following these steps should help fix the problem on the 20th for anyone who needs to also update the RSA keys.
